server/handlers/wrappers.go

```diff
@@ -24,6 +28,19 @@ func AsUser(handler UserHandler) gin.HandlerFunc {
 	}
 }
 
+// AsUserWithAppConfig wraps the handler with the user context. User can be authorized by xPub or accessKey
+// it also passes appConfig to the handler
+func AsUserWithAppConfig(handler UserHandlerWithAppConfig, appConfig *config.AppConfig) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		userContext := reqctx.GetUserContext(c)
+		if userContext.GetAuthType() == reqctx.AuthTypeAdmin {
+			spverrors.AbortWithErrorResponse(c, spverrors.ErrAdminAuthOnUserEndpoint, nil)
+			return
+		}
+		handler(c, appConfig)
+	}
+}
```

Comment:
This is not needed see my comment above ☝️

Reply:
removed